Kernel/FileSystem/FATFileSystem.h

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2022, Undefine <undefine@undefine.pl>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/OwnPtr.h>
+#include <AK/RefPtr.h>
+#include <AK/Types.h>
+#include <Kernel/FileSystem/BlockBasedFileSystem.h>
+#include <Kernel/FileSystem/Inode.h>
+#include <Kernel/KBuffer.h>
+
+namespace Kernel {
+
+struct [[gnu::packed]] FAT32BootRecord {
+    u8 boot_jump[3];
+    u8 oem_identifier[8];
+    u16 bytes_per_sector;
+    u8 sectors_per_cluster;
+    u16 reserved_sector_count;
+    u8 fat_count;
+    u16 root_directory_entry_count;
+    u16 unused;
+    u8 media_descriptor_type;
+    u16 unused2;
+    u16 sectors_per_track;
+    u16 head_count;
+    u32 hidden_sector_count;
+    u32 sector_count;
+    u32 sectors_per_fat;
+    u16 flags;
+    u16 fat_version;
+    u32 root_directory_cluster;
+    u16 fs_info_sector;
+    u16 backup_boot_sector;
+    u8 reserved[12];
+    u8 drive_number;
+    u8 reserved2;
+    u8 signature;
+    u32 volume_id;
+    u8 volume_label_string[11];
+    u8 system_identifier_string[8];
+};
+static_assert(sizeof(FAT32BootRecord) == 90);
+
+struct [[gnu::packed]] FATEntry {
+    u8 filename[8];
+    u8 extension[3];
+    u8 attributes;
+    u8 reserved;
+    u8 creation_time_seconds;
+    u16 creation_time;
+    u16 creation_date;
+    u16 last_accessed_date;
+    u16 first_cluster_high;
+    u16 modification_time;
+    u16 modification_date;
+    u16 first_cluster_low;
+    u32 file_size;
+};
+static_assert(sizeof(FATEntry) == 32);
+
+struct [[gnu::packed]] FATLongFileNameEntry {
+    u8 entry_index;
+    u16 characters1[5];
+    u8 attribute;
+    u8 entry_type;
```

Comment:
Is there an `enum` (or `enum`-like) type to go with this? (I do realize that the entry type seems to be currently unused by our code.)

Reply:
The documentation says that it's always zero.